idaes/generic_models/properties/core/generic/generic_property.py

```diff
@@ -716,19 +715,18 @@ def build(self):
                         # First validate that p is a phase
                         if p not in self.phase_list:
                             raise ConfigurationError(
-                                f"{self.name} property {prop} definition contained"
-                                f" unrecognised phase {p}.")
+                                f"{self.name} property {prop} definition "
+                                f"contained unrecognised phase {p}.")
                         else:
                             try:
                                 meth.build_parameters(cobj, p)
                             except AttributeError:
```

Comment:
Why try-except instead of 
```python
if hasattr(meth, 'build_parameters'):
    # It is not required that there be a 'build_parameters' methods
    meth.build_parameters(cobj, p)
```

Reply:
Mainly because I expect the method to be there more often that not. From my understanding, `try/except` will be faster when the component is there, but is more expensive to handle the exception that occurs when it isn't. `hasattr`  on the other hand has as constant overhead of the `hasattr` call. So, if you expect the attribute to be there, `try/except` should be faster.

Reply:
It is, but only slightly:
```python
from timeit import timeit

tests = [
    "try:\n a.foo()\nexcept:\n pass",
    "try:\n b.foo()\nexcept:\n pass",
    "if hasattr(a, 'foo'):\n a.foo()",
    "if hasattr(b, 'foo'):\n b.foo()",
]

class A():
    def foo(self):
        pass

class B():
    pass


for test in tests:
    print("%7.3f %s" % (
        timeit(test, 'from __main__ import A,B; a=A(); b=B()'),
        test.replace('\n', '\\n')))
```
gives
```
 python26  python27 python310  python33  python34  python35  python36  python37  python38  python39
    2.6.9    2.7.18    3.10.0     3.3.7    3.4.10    3.5.10    3.6.15    3.7.12    3.8.12     3.9.8
    0.100     0.070     0.059     0.083     0.089     0.086     0.070     0.057     0.060     0.058   try:\n a.foo()\nexcept:\n pass
    0.346     0.320     0.301     0.257     0.246     0.226     0.226     0.224     0.227     0.222   try:\n b.foo()\nexcept:\n pass
    0.139     0.120     0.101     0.116     0.128     0.115     0.127     0.098     0.095     0.109   if hasattr(a, 'foo'):\n a.foo()
    0.219     0.188     0.039     0.205     0.203     0.170     0.172     0.041     0.037     0.040   if hasattr(b, 'foo'):\n b.foo()
```